src/bin/casr-python.rs

```diff
@@ -0,0 +1,296 @@
+
+    // Create report.
+    let mut report = CrashReport::new();
+    report.executable_path = argv[0].to_string();
+    report.proc_cmdline = argv.join(" ");
+    let _ = report.add_os_info();
+
+    // Get python report.
+    let python_stderr_list: Vec<String> =
+        python_stderr.split('\n').map(|l| l.to_string()).collect();
+
+    if python_stderr_list
+        .iter()
+        .any(|line| line.contains("ERROR: libFuzzer: "))
+    {
+        let python_stdout = String::from_utf8_lossy(&python_result.stdout);
+        let python_stdout_list: Vec<String> =
+            python_stdout.split('\n').map(|l| l.to_string()).collect();
+
+        if let Some(report_start) = python_stdout_list
+            .iter()
+            .position(|line| line.contains("Uncaught Python exception: "))
+        {
+            // Set python report in casr report.
+            let report_end = python_stdout_list
+                .iter()
+                .rposition(|s| !s.is_empty())
+                .unwrap()
+                + 1;
+            report.python_report = Vec::from(&python_stdout_list[report_start..report_end]);
+
+            // Get exception from python report.
+            let re = Regex::new(r#"([\w]+): (.+)"#).unwrap();
+            if let Some(cap) = re.captures(&report.python_report[1]) {
+                report.execution_class.severity = Cow::Borrowed("UNDEFINED");
+                report.execution_class.short_description =
+                    Cow::Borrowed(cap.get(1).unwrap().as_str());
+                report.execution_class.description = Cow::Borrowed(cap.get(2).unwrap().as_str());
+                report.execution_class.explanation = Cow::Borrowed("");
+            }
+
+            // Get stack trace from python report.
+            let first = report
+                .python_report
+                .iter()
+                .position(|line| line.starts_with("Traceback "));
+            if first.is_none() {
+                bail!("Couldn't find traceback in python report");
+            }
+
+            // Stack trace is splitted by empty line.
+            let first = first.unwrap();
+            let last = report
+                .python_report
+                .iter()
+                .skip(first)
+                .rposition(|s| !s.is_empty());
+            if last.is_none() {
+                bail!("Couldn't find traceback end in python report");
+            }
+            let last = last.unwrap();
+
+            let re =
+                Regex::new(r#"(File "[\w\-<>~./]+", line [\d]+, in [\w<>]+|\[Previous line repeated (\d+) more times\])"#).unwrap();
+            report.stacktrace = report.python_report[first..first + last]
+                .iter()
+                .rev()
+                .map(|s| s.trim().to_string())
+                .filter(|s| re.is_match(s))
+                .collect::<Vec<String>>();
+        } else {
+            bail!("Need to use casr-san with LD_PRELOAD")
```

Comment:
Error is unclear. Tool is called  casr-python, why  casr-san?

Reply:
Cuz this case means there is a crash, but in C/C++ extension - from sanitizers, so we need to call casr-san.
Later I will write auto casr-san call here.

Reply:
Add casr-san launch